src/internal/base-component/index.ts

```diff
@@ -39,9 +39,11 @@ export function getBaseProps(props: BaseComponentProps) {
 }
 
 export interface BasePropsWithAnalyticsMetadata {
+  analyticsMetadata?: AnalyticsMetadata;
   __analyticsMetadata?: AnalyticsMetadata;
 }
 
 export function getAnalyticsMetadataProps(props?: unknown) {
-  return (props as BasePropsWithAnalyticsMetadata | undefined)?.__analyticsMetadata;
+  const propsWithAnalytics = props as any as BasePropsWithAnalyticsMetadata;
```

Comment:
Do we still need the unsafe type assertion? Now that each component has the `analyticsMetadata` property in its public types, we could replace `unknown` in the parameter with something more strict like `BasePropsWithAnalyticsMetadata`

Reply:
Changed